http/src/handler.rs

```diff
@@ -562,10 +616,58 @@ impl<M: Metadata, S: Middleware<M>> RpcHandler<M, S> {
 	/// Returns true if the `content_type` header indicates a valid JSON
 	/// message.
 	fn is_json(content_type: Option<&header::HeaderValue>) -> bool {
-		match content_type.and_then(|val| val.to_str().ok()) {
-			Some("application/json") => true,
-			Some("application/json; charset=utf-8") => true,
+		match content_type.and_then(|val| val.to_str().map(str::to_lowercase).ok()) {
+			Some(ref content)
+				if content == "application/json"
+					|| content == "application/json; charset=utf-8" =>
+			{
+				true
+			}
 			_ => false,
 		}
 	}
 }
+
+#[cfg(test)]
+mod test {
+	use super::{hyper, RpcHandler};
+	use jsonrpc_core::futures::future::Either;
+	use jsonrpc_core::futures::Future;
+	use jsonrpc_core::*;
+	use std::sync::atomic::{AtomicUsize};
+	
+	#[derive(Clone, Debug)]
+	struct Meta(usize);
+	
+	impl Metadata for Meta {}
+
+	#[derive(Default)]
+	struct MyMiddleware(AtomicUsize);
+
+	impl Middleware<Meta> for MyMiddleware {
+		type Future = FutureResponse;
+		type CallFuture = middleware::NoopCallFuture;
+
+		fn on_request<F, X>(&self, request: Request, meta: Meta, next: F) -> Either<Self::Future, X>
+		where
+			F: FnOnce(Request, Meta) -> X + Send,
+			X: Future<Item = Option<Response>, Error = ()> + Send + 'static,
+		{
+			Either::A(Box::new(next(request, meta)))
+		}
+	}
+
+
+	#[test]
+	fn test_case_insensitive_content_type() {
+		let request = hyper::Request::builder()
```

Comment:
Can we please add another assertion to make sure that the builder doesn't normalize the value?